### ISSUE TITLE
This ... there are many hackers on OpenSpades servers

### ISSUE THREAD

**Issue by xXxToxicBlueDustxXx** [yvt/openspades#496]
I think this someone posted it but ...

Lately I have noticed that in CASI all servers that have buildandshot have at least 1 or 2 hackers the problem that not having an anti-hack system refuses to use the command /admin to send a message to the administrator and from there it resides there That nobody does it then .. on youtube there are already several hacks that are:

- AIMBOT that this is the most used in CTF
- Wall Hack
- use cliché exploits
Ects

Then I do not know what will happen for the next update this is already critical that has to do with the protection of the server but if they give you something to do when the clients say the program Discover that one of your dlls is modified or some other program is injecting A process Example: cheatengine trying to make hacks via hexagecimal editing but not reporting false positive for popular crankers like fraps mirillis action etc

Could give an idea that if the customer decta that has been modified the fundamental scripts do not arrenque or something similar but if it was compiled in debug mode or in another form there is left because it is in development mode

I hope a sincere answer and to see that it will separate us in the next update

**Comment by feikname** [comment]
The issue you are talking abuut is [this one](https://github.com/yvt/openspades/issues/463). (#463)

**Comment by VelocityRa** [comment]
@xXxToxicBlueDustxXx Like I explain in detail in the issue mentioned above:
**An anticheat is literally impossible with the current server protocol.**
Even the Admin/Report tools idea I explain [here](https://github.com/yvt/openspades/issues/463#issuecomment-263569742) needs server-side changes.

**Comment by xXxToxicBlueDustxXx** [comment]
there is a way if the server banned and user the server will write the ip into a banlist ( seems like The source engine handle this ) or in mayor instances block the mac (means to no acces a any server)

**Comment by VelocityRa** [comment]
>the server will write the ip into a banlist

Yes... that would require server-side changes, thus it does not belong in the OpenSpades issue tracker.

**Comment by xXxToxicBlueDustxXx** [comment]
Apart it would be necessary to more people that work in that part could not be I give rune idea to create a service of hosting apart from the one that uses openspades one that is separated to that something like serverlist.openspades.com possible is but could be done later

**Comment by feikname** [comment]
@xXxToxicBlueDustxXx That's a issue planned to be dealt with in the next-gen protocol project. However, the development is abandoned for now.

**Comment by xXxToxicBlueDustxXx** [comment]
But if a person outside the projection does that it would be possible to be accepted? I say I have this intention to do it alone I look for a good place to do it

**Comment by feikname** [comment]
@xXxToxicBlueDustxXx I didn't quite grasp the concept. What exatctly is your idea?

**Comment by xXxToxicBlueDustxXx** [comment]
My idea would be to separate Openspades from buildandshot.com by doing a page where the servers (if possible having the steam works sdk) that when creating it automatically these are protected an example of this is H1Z1

**Comment by xXxToxicBlueDustxXx** [comment]
In short would be aglo like:

- when creating the server automatically one can decide whether one wants the VAC or not
- whether it is private or public or a lobby (server that is hosted throughout the year by a page or group)
- have the client check before starting if all essential files are not modified
- and finally the most important one that runs steam

This idea found in the forums steam works saying that if the vac could protect servers that were created online and it turns out that it is possible only a medium advanced level of SQL

**Comment by feikname** [comment]
@xXxToxicBlueDustxXx your idea sure is possible, but it would require radical changes, probably some money, and a lot of development time. The anti-cheat integration is planned to be dealt with as soon as next-gen protocol development returns. (My hopes are for after 0.1.1, in the middle of 2017).

**The focus for now is on new features, enchancements, bug fixes and ensuring compatibility.** When we get that done, next-gen development will probably return. Bringing OpenSpades to Steam, implementing an anti-cheat in the new servers and ensuring compatibility with the old ones seems like the the way to go to increase the number of players while also keeping the old ones. 

I have no doubts that many abandoned Ace of Spades classic and OpenSpades clients because the servers almost always have cheaters. But It's been like that for a long time already, I think it cant wait a bit more.

**Perhaps improving pySnip and pySpades with better anti-cheat integration and convincing the current server owners to upgrade would be the best solution for now.**

Also, this issue is a duplicate of #463, so I am closing it.
